include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -855,76 +864,117 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUImage** const _begin,
             barrier.dstQueueFamilyIndex = transferFamIx;
             barrier.barrier.srcAccessMask = asset::EAF_TRANSFER_READ_BIT;
             barrier.barrier.dstAccessMask = asset::EAF_TRANSFER_WRITE_BIT;
-            cmdbuf_transfer->pipelineBarrier(asset::EPSF_TRANSFER_BIT, asset::EPSF_TRANSFER_BIT, 0, 0u, nullptr, 1u, &barrier, 0u, nullptr);
-            cmdbuf_transfer->copyBufferToImage(buf.get(), img, asset::EIL_UNDEFINED, cpuimg->getRegions().size(), cpuimg->getRegions().begin());
+
+            IGPUCommandBuffer::SImageMemoryBarrier toTransferDst = {};
+            toTransferDst.barrier.srcAccessMask = static_cast<asset::E_ACCESS_FLAGS>(0u);
+            toTransferDst.barrier.dstAccessMask = asset::EAF_TRANSFER_WRITE_BIT;
+            toTransferDst.oldLayout = asset::EIL_UNDEFINED;
+            toTransferDst.newLayout = asset::EIL_TRANSFER_DST_OPTIMAL;
+            toTransferDst.srcQueueFamilyIndex = transferFamIx;
```

Comment:
who did the image belong to before? (you just created it really)

Reply:
should be mem barrier on buffer, gonna fix this